### DISCUSSION TITLE
Vips jpegsave has no bitdepth option

### DISCUSSION THREAD

**Discussion by SangNguyen99** [libvips/libvips#3847] [General]
The official site document said bitdepth option is available on all saver and loader but apparently when I run vips jpegsave there is no bitdepth option and also trying to jpegsave with bitdepth option throws an error. Here are my installed version and related jpeg dependencies: vips-8.15.1, libexif-0.6.22-2.el7_9.x86_64, libjpeg-turbo-1.2.90-8.el7.x86_64, openjpeg-1.5.1-18.el7.x86_64.
Here is the command that I ran:  vipsthumbnail source.jpg -s 500 -o 'bench.jpg[Q=75,background=0,keep=none,bitdepth=8]
' --vips-info

**Comment by jcupitt** [comment]
Hello @SangNguyen99,

The JPEG saver will automatically convert to 8 bit on save, you don't need to set any options.

libjpeg can only save 8 and 12 bit data, and this must be configured at compile time -- you can't have 8 and 12 bit JPEG save in the same binary without some crazy tricks. And I think almost no one will be able to view 12 bit JPEGs anyway, so it's only useful in some very niche applications.

**Reply by SangNguyen99** [reply]
Hi John,
I am working on migrating ImageMagick to vips and apparently they use the same compile time configuration for their default release so their -depth option was just a suggestion for jpeg. Thanks for answering, I love the work you guys been doing with vips and I will close the discussion then.
